### ISSUE TITLE
lfs.lock_dir really a good idea?

### ISSUE THREAD

**Issue by rrthomas** [keplerproject/luafilesystem#9]
Locking is a complicated subject, and lfs.lock_dir implements its own mechanism. It's about 100 lines of C, and it's far from clear that it's useful or robust. It is hard to tell whether the behaviour is the same on Window and non-Windows system (it doesn't look like it). There are no tests for it. Something similar could easily be implemented using lfs.lock in a few lines of Lua. I suggest it be removed, or, failing that, tests be added. Contrary to the documentation, there seems to be no second parameter to check for staleness either on Windows or non-Windows systems (if a second parameter is passed, it is ignored).

In summary, this has the air of application-specific code that is not ready to be put in a library.
